src/Parallel/Actions/SetupDataBox.hpp

```diff
@@ -0,0 +1,132 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <tuple>
+#include <type_traits>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "Parallel/GlobalCache.hpp"
+#include "ParallelAlgorithms/Initialization/MergeIntoDataBox.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace Actions {
+/// \cond
+struct SetupDataBox;
+/// \endcond
+
+namespace detail {
+CREATE_HAS_TYPE_ALIAS(simple_tags)
+CREATE_HAS_TYPE_ALIAS_V(simple_tags)
+CREATE_HAS_TYPE_ALIAS(compute_tags)
+CREATE_HAS_TYPE_ALIAS_V(compute_tags)
+
+template <typename Action, typename enable = std::bool_constant<true>>
```

Comment:
You can do this without `has_type_alias`. In this line, replace the `enable` parameter with `typename = std::void_t<>` and immediately define this struct with the `type = tmpl::list<>`. Then, in the specialization below, replace the `std::bool_constant` with `std::void_t<typename Action::simple_tags>`. SFINAE will do the right thing :) The same applies to the compute tags of course.